### PR TITLE
feat(studio): use eslint, prettier, and tsconfig tooling packages in studio

### PR DESCRIPTION
**Pull request by karrui**
### TL;DR

This PR replaces the current lint and formatter used in isomer studio with the eslint and prettier tooling packages, in order to maintain consistency of linting rules and formatting across the whole monorepo.

### What changed?

- Removed root `.prettierrc.json` file so prettier settings in repo is respected.
- Refactored existing codebase to conform with the new linting and formatting rules enforced by ESLint and Prettier.

### Why make this change?

The integration of Prettier with ESLint eliminates the need for a separate Prettier configuration file and ensures that code formatting rules are consistently applied across the entire codebase by using a single tool, ESLint, for both linting and formatting.

### CODE REVIEW

**Comment by karrui** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/281?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#282** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/282?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#281** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/281?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#280** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/280?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#279** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/279?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @karrui and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/studio/package.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "version": "0.0.1",
   "private": true,
   "scripts": {
-    "generate": "prisma generate",
+    "generate": "prisma generate && npm run lint:fix && npm run format:fix",
```
Comment (seaerchin) [inline apps/studio/package.json:6]: nit: would prefer these to be in 2 separate commands (ie, we keep the current `generate` and have 1 extra command called `format:fix` or smth) but more a style preference than anything

**Review by seaerchin — APPROVED** [review]
i skimmed alot (read: most) of the files cos i realised they were formatting/import changes.

please dismiss if not

**Comment by karrui** [comment]
### Merge activity

* **Jul 9, 6:24 AM EDT**: @karrui started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/opengovsg/isomer/281).

**Pull request merged**